components/Pagination/react.js

```diff
@@ -0,0 +1,120 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import classNames from 'classnames/bind'
+import { React as Button } from 'components/Button'
+import { generateVisiblePages } from './generateVisiblePages'
+
+import styles from './style.module.scss'
+
+const cx = classNames.bind(styles)
+
+const Pagination = ({
+  page,
+  totalItems,
+  itemsPerPage,
+  onChange,
+  prevLabel,
+  nextLabel,
+  viewingLabel,
+}) => {
+  // Calculate total pages needed & generate an array of page numbers.
```

Comment:
Damn this is :100: and very clean. So clean that a lot of these comments are redundant, as the variables are self describing.

Example...

// total pages 
const totalPages

// our prev / next page
const prevPage, nextPage

// Return an unclickable ellipsis.
<span>not an anchor tag</span>

It'd reduce the LOC, which could help read.

Reply:
@tarikhamilton I agree that some of these comments might be overkill and removed them.

FWIW visually breaking up longer declarations blocks up like this with a single line comments IMO improves readability.